lib/models/proposal.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:convert';
+
+import 'package:date_format/date_format.dart';
+
+class Proposal {
+  final int pid;
+  final String title;
+  final String description;
+  final String endTime;
+  final int uid;
+  final int yesVotes;
+  final int noVotes;
+  final DateTime date;
+
+  Proposal({
+    this.pid,
+    this.title,
+    this.description,
+    this.endTime,
+    this.uid,
+    this.yesVotes,
+    this.noVotes,
+    this.date,
+  });
+
+  factory Proposal.fromJson({
+    Map<String, dynamic> map,
+    String prefLangCode,
+  }) {
+    DateTime endTime = DateTime.parse(map['endtime']);
+    String endTimeParsed = formatDate(
+        endTime, [mm, '/', dd, '/', yyyy, ' ', hh, ':', nn, ' ', am]);
+    String translatedTitle = json.decode(map['title'])[prefLangCode];
+    String translatedDescription =
+        json.decode(map['description'])[prefLangCode];
+    return Proposal(
+      pid: map['pid'],
+      title: translatedTitle,
+      description: translatedDescription,
+      endTime: endTimeParsed,
+      uid: map['uid'],
+      yesVotes: map['numyes'],
+      noVotes: map['numno'],
+      date: endTime,
```

Comment:
are both endTimeParsed and endTime needed ? it's two representations of the same info

Reply:
this was also left over from before

Reply:
turns out we do use it to calculate the difference in days left in proposal cards.

resolve / comment